adapta/storage/distributed_object_store/datastax_astra/_models.py

```diff
@@ -0,0 +1,137 @@
+"""
+ Auxiliary models for Astra client operations.
+"""
+from functools import reduce
+from typing import final, List, Dict, Generic, TypeVar, Any, Union
+
+TField = TypeVar("TField")  # pylint: disable=invalid-name
+
+
+@final
+class AstraFilterExpression:
+    """
+    Filter expression from combining two or more Astra fields
+    """
+
+    def __init__(self, expr: List[Dict[str, Any]]):
+        """
+        Creates an AstraFilterExpression
+        """
+        self._expr = expr
+
+    @property
+    def expression(self) -> List[Dict[str, Any]]:
+        """
+        Generated expression.
+        """
+        return self._expr
+
+    def __and__(self, other: Union["AstraFilterExpression", "AstraField"]) -> "AstraFilterExpression":
+        added_expr = [other.expression] if isinstance(other, AstraField) else other.expression
+        self._expr = reduce(lambda e1, e2: e1 | e2, self._expr + added_expr)
+        return self
+
+    def __or__(self, other: Union["AstraFilterExpression", "AstraField"]) -> "AstraFilterExpression":
+        added_expr = [other.expression] if isinstance(other, AstraField) else other.expression
+        self._expr = self._expr + added_expr
+        return self
+
+
+@final
+class AstraField(Generic[TField]):
+    """
+    Represents a field in the Astra table. Semantics mimic pyarrow.
+    Based on https://docs.datastax.com/en/developer/python-driver/3.24/cqlengine/queryset/
+    """
+
+    def __init__(self, field_name: str):
+        """
+        Creates an instance of AstraField with empty filters applied.
+        """
+        self._field_name = field_name
+        self._field_filters: Dict[str, TField] = {}
+
+    @property
+    def field_name(self):
+        """
+        Name of the wrapped field.
+        """
+        return self._field_name
+
+    def with_filters(self, field_filters: Dict[str, TField]) -> "AstraField[TField]":
+        """
+        Merge current filters with provided ones.
+        """
+        self._field_filters.update(field_filters)
+        return self
+
+    def isin(self, values: List[TField]) -> "AstraField[TField]":
+        """
+        Generates a filter condition checking that field value is one of the values provided.
+        """
+        self._field_filters.update({f"{self._field_name}__in": values})
```

Comment:
I am not a fan of relying on string operations to access the operator. Can we use `Dict[str, Tuple[str, TField]]` or `Dict[str, Dict[str, str]]` instead?

Reply:
not sure what you mean here tbf :)

Reply:
Let's say I do following, `AstraField('col1').isin([1,2,3])`, this would be represented as `{"col1__in": [1,2,3]}`. In order to know what operator is used, I need to do something like `isin_filters = {(col_op, val) for col_op, val in self._filters if col_op.endswith("__in")}` (thought scenario). I do not like I need to check for substrings to check the operator, like the isin operator. I hope this makes more sense